community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/BuiltInProcedures.java

```diff
@@ -162,8 +164,11 @@
                         type = IndexType.NODE_LABEL_PROPERTY.typeName();
                     }
 
-                    result.add( new IndexResult( "INDEX ON " + index.schema().userDescription( tokens ),
-                            operations.indexGetState( index ).toString(), type ) );
+                    String label = tokens.labelGetName( index.schema().getLabelId() );
+                    List<String> propertyNames = propertyNames( tokens, index );
+                    result.add( new IndexResult( "INDEX ON " + index.schema().userDescription( tokens ), label, propertyNames,
```

Comment:
not sure why we need label and propertyNames once again since they are already part of `index.schema().userDescription( tokens )` message

Reply:
Request from field to make it easier to access programmatically. This is also the reason for returning provider information as a map.